006_team_4_agent_design/006_team_4_agent_design.tex

```diff
@@ -1 +1,311 @@
+
+\begin{figure}[htb]
+    \centering
+    \includegraphics{006_team_4_agent_design/assets/evolution_overview.png}
+    \caption{Overview of evolutionary algorithm.}
+    \label{fig:evolutionOverview}
+\end{figure}
+
+\section{Evolutionary Algorithm}
+
+In the fundamentals of evolutionary algorithms, every agent has a set of genes (or genotypes), which can express themselves (as phenotypes) in the way the agent interacts with its surroundings. Therefore, the aim is to find the optimal genotypes that will give phenotypes that are deemed to be favourable. The way in which this is implemented is by seeing the performance of an agent according to a particular metric. This way the effect of each phenotype (and therefore underlying genotype) is put into perspective and ranked against the others. The most successful agents survive, while the less performant ones are killed off. When the surviving population reproduces, the resulting offspring will share the genes of both their parents, meaning they will have a mix of the most successful genotypes. Furthermore, there are some offspring with mutated genes and some that have identical genes to the previous generation for genetic variation. With this new population, the process is repeated to iteratively create more successful populations, as illustrated in \Cref{fig:team4evolve}.
+
+\begin{figure}[htb]
+    \centering
+    \includegraphics[width=0.7\textwidth]{006_team_4_agent_design/assets/evolve.png}
+    \caption{Population selection process in an evolutionary algorithm.}
+    \label{fig:team4evolve}
+\end{figure}
+
+Hence, with the use of an evolutionary algorithm, the final agent was the culmination of multiple generations of populations, with the genes of successful ancestors having been propagated and mixed. This meant it was a well-adapted agent which was robust to the many environments it could encounter in the tower. 
+
+\subsection{Agent Genotypes}
+To use evolutionary algorithms optimally, the genotypes that evolve should be well defined. This section outlines the two genotypes that were chosen as they were deemed to have the most effect on the survival of an agent in the system.
+
+\subsubsection{FoodtoEat}\label{FoodtoEat}
+\texttt{FoodToEat} was the amount of food the agent would eat on any given day of the simulation. Given the HP update curves in \Cref{updateHP}, it can be noted that any consumption past the threshold of 60 food had little to no effect on the HP gain of the agent. This meant that eating any more than that amount was wasteful and sub-optimal, so during the training process the algorithm was capped. This meant that the process could never learn to eat a value above 60, just as it could not learn to eat a value below 0.
```

Comment:
might need to use `\textit{}` rather than `\texttt{}` depending on the final decision of the editors

Reply:
What was the final decision regarding this? Will update inline with that

Reply:
texttt is the final decision, sorry for the late answer